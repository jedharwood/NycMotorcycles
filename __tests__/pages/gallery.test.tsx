import React from 'react';

import { render, screen, waitFor } from '@testing-library/react';
import { useRouter } from 'next/router';
import { useIntl } from 'react-intl';

import GalleryPage from '@/pages/sold-archive/gallery/[bike]';

jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));
jest.mock('react-intl', () => ({
    useIntl: jest.fn(),
}));

describe('GalleryPage', () => {
    const closeImageModalMock = jest.fn();

    beforeEach(() => {
        jest.spyOn(React, 'useContext').mockImplementation(() => ({
            closeImageModal: closeImageModalMock,
        }));
        (useRouter as jest.Mock).mockReturnValue({
            isReady: false,
        });
        (useIntl as jest.Mock).mockReturnValue({
            formatMessage: ({ id }: { id: string }) => id,
        });
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should render the spinner until the router is ready', async () => {
        render(<GalleryPage />);

        const spinner: HTMLElement = await screen.findByTestId(
            'loading-spinner',
            {},
            { timeout: 2000 }
        );
        expect(spinner).toBeInTheDocument();
    });

    it('should render the kawasaki-h1ra gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'kawasaki-h1ra' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText('pg.gallery.kawasaki-h1ra.name');
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render the harley-xr750-75 gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'harley-xr750-75' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText(
                'pg.gallery.harley-xr750-75.name'
            );
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render the ducati-250 gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'ducati-250' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText('pg.gallery.ducati-250.name');
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render the honda-dream gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'honda-dream' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText('pg.gallery.honda-dream.name');
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render the harley-xr750-replica gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'harley-xr750-replica' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText(
                'pg.gallery.harley-xr750-replica.name'
            );
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });

    it('should render the harley-xr750 gallery page', async () => {
        (useRouter as jest.Mock).mockReturnValue({
            isReady: true,
            query: { bike: 'harley-xr750' },
        });
        const { container } = render(<GalleryPage />);

        await waitFor(() => {
            const legend: HTMLElement = screen.getByText('pg.gallery.harley-xr750.name');
            expect(legend).toBeInTheDocument();
        });
        expect(container).toMatchSnapshot();
    });
});
