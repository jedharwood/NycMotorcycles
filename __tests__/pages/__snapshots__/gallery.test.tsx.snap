// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryPage should render the ducati-250 gallery page 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="pg.sold-archive.alt.ducati-250"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            pg.gallery.ducati-250.name
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <section
          class="space-y-4"
        >
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.ducati-250.alt.pitstop"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.ducati-250.alt.graf"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.ducati-250.alt.garden"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.ducati-250.alt.tank"
            />
          </div>
        </section>
        <div
          class="space-y-4"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                pg.gallery.ducati-250.title
              </h2>
            </div>
            <ul
              class="space-y-2"
            >
              <li>
                pg.gallery.ducati-250.list-1
              </li>
              <li>
                pg.gallery.ducati-250.list-2
              </li>
              <li>
                pg.gallery.ducati-250.list-3
              </li>
              <li>
                pg.gallery.ducati-250.list-4
              </li>
              <li>
                pg.gallery.ducati-250.list-5
              </li>
              <li>
                pg.gallery.ducati-250.list-6
              </li>
              <li>
                pg.gallery.ducati-250.list-7
              </li>
              <li>
                pg.gallery.ducati-250.list-8
              </li>
              <li>
                pg.gallery.ducati-250.list-9
              </li>
              <li>
                pg.gallery.ducati-250.list-10
              </li>
              <li>
                pg.gallery.ducati-250.list-11
              </li>
              <li>
                pg.gallery.ducati-250.list-12
              </li>
              <li>
                pg.gallery.ducati-250.list-13
              </li>
              <li>
                pg.gallery.ducati-250.list-14
              </li>
              <li>
                pg.gallery.ducati-250.list-15
              </li>
            </ul>
            <div
              class="space-y-4 undefined"
            >
              <p>
                pg.gallery.ducati-250.para-1
              </p>
              <p>
                pg.gallery.ducati-250.para-2
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`GalleryPage should render the harley-xr750-75 gallery page 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="pg.sold-archive.alt.harley-xr750-75"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            pg.gallery.harley-xr750-75.name
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <section
          class="space-y-4"
        >
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-75.alt.showroom"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-75.alt.pier"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-75.alt.beach"
            />
          </div>
        </section>
        <div
          class="space-y-4"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                pg.gallery.harley-xr750-75.title
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                pg.gallery.harley-xr750-75.para-1
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-2
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-3
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-4
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-5
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-6
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-7
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-8
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-9
              </p>
              <p>
                pg.gallery.harley-xr750-75.para-10
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`GalleryPage should render the harley-xr750-replica gallery page 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="pg.sold-archive.alt.harley-xr750-replica"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            pg.gallery.harley-xr750-replica.name
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <section
          class="space-y-4"
        >
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-replica.alt.right-side"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-replica.alt.graf"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-replica.alt.rider"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.harley-xr750-replica.alt.larry"
            />
          </div>
        </section>
        <div
          class="space-y-4"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                pg.gallery.harley-xr750-replica.sect-1.title
              </h2>
            </div>
            <div
              class="space-y-4 undefined"
            >
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-1
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-2
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-3
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-4
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-5
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-6
              </p>
              <p>
                pg.gallery.harley-xr750-replica.sect-1.para-7
              </p>
            </div>
          </section>
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                pg.gallery.harley-xr750-replica.sect-2.title
              </h2>
            </div>
            <ul
              class="space-y-2"
            >
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-1
              </li>
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-2
              </li>
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-3
              </li>
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-4
              </li>
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-5
              </li>
              <li>
                pg.gallery.harley-xr750-replica.sect-2.list-6
              </li>
            </ul>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`GalleryPage should render the honda-dream gallery page 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="pg.sold-archive.alt.honda-dream"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            pg.gallery.honda-dream.name
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <section
          class="space-y-4"
        >
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.honda-dream.alt.showroom"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.honda-dream.alt.harbour"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.honda-dream.alt.rear-centre"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.honda-dream.alt.right-side"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.honda-dream.alt.rear"
            />
          </div>
        </section>
        <div
          class="space-y-4"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="text-center "
            >
              <h2
                class="text-xl font-medium opacity-80 md:text-2xl"
              >
                pg.gallery.honda-dream.title
              </h2>
              <p>
                pg.gallery.honda-dream.subtitle
              </p>
            </div>
            <ul
              class="space-y-2"
            >
              <li>
                pg.gallery.honda-dream.list-1
              </li>
              <li>
                pg.gallery.honda-dream.list-2
              </li>
              <li>
                pg.gallery.honda-dream.list-3
              </li>
              <li>
                pg.gallery.honda-dream.list-4
              </li>
              <li>
                pg.gallery.honda-dream.list-5
              </li>
            </ul>
            <div
              class="space-y-4 undefined"
            >
              <p>
                pg.gallery.honda-dream.para-1
              </p>
              <p>
                pg.gallery.honda-dream.para-2
              </p>
            </div>
            <h3
              class="text-center text-xl font-medium opacity-80 md:text-2xl"
            >
              pg.gallery.honda-dream.footer
            </h3>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`GalleryPage should render the kawasaki-h1ra gallery page 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="pg.sold-archive.alt.kawasaki-h1ra"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            pg.gallery.kawasaki-h1ra.name
          </h2>
        </div>
      </div>
      <div
        class="grid gap-4 md:grid-cols-2"
      >
        <section
          class="space-y-4"
        >
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.kawasaki-h1ra.alt.left-side"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.kawasaki-h1ra.alt.fairing"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.kawasaki-h1ra.alt.bars"
            />
          </div>
          <div
            class="rounded-md
            card-hover cursor-pointer 
            h-fit"
          >
            <img
              alt="pg.gallery.kawasaki-h1ra.alt.tank"
            />
          </div>
        </section>
        <div
          class="space-y-4"
        >
          <section
            class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
          >
            <div
              class="space-y-4 undefined"
            >
              <p>
                pg.gallery.kawasaki-h1ra.para-1
              </p>
              <p>
                pg.gallery.kawasaki-h1ra.para-2
              </p>
              <p>
                pg.gallery.kawasaki-h1ra.para-3
              </p>
              <p>
                pg.gallery.kawasaki-h1ra.para-4
              </p>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
